packages/registry/middleware/middleware_test.go

```
@@ -0,0 +1,194 @@
+	h := &Handler{
+		next:         http.NotFoundHandler(),
+		serviceToken: "service-token",
+		garbageCollect: func(_ context.Context, organizationID string, jobID string) (garbagecollection.Report, error) {
+			if organizationID != "org-1" || jobID != "job-1" {
+				t.Fatalf("organization=%q job=%q", organizationID, jobID)
+			}
+			called = true
+			return garbagecollection.Report{BytesBefore: 100, BytesAfter: 25}, nil
+		},
+	}
+	request := httptest.NewRequest(http.MethodPost, "/internal/organizations/org-1/garbage-collection", nil)
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)